ads/google/imaVideo.js

```diff
@@ -221,241 +195,154 @@ let consentState;
 // Throttle for the showControls() function
 let showControlsThrottled = throttle(window, showControls, 1000);
 
+/**
+ * @param {!Node} parent
+ * @param {string} css
+ */
+function insertCss(parent, css) {
+  const style = parent.ownerDocument.createElement('style');
+  style./*OK*/ textContent = css;
+  parent.appendChild(style);
+}
+
+/**
+ * @param {string} state
+ * @param {boolean} active
+ */
+function toggleRootDataAttribute(state, active) {
+  const {'root': root} = elements;
+  const attributeName = `data-${state}`;
+  if (active) {
+    root.setAttribute(attributeName, '');
+  } else {
+    root.removeAttribute(attributeName);
+  }
+}
+
+/**
+ * @param {Document} elementOrDoc
+ * @return {!Object<string, !Element>}
+ */
+function renderElements(elementOrDoc) {
+  const html = htmlFor(elementOrDoc);
+
+  // Elements annotated with `ref="name"` are referenced as `elements['name']`.
+  // They also have their `ref` copied as a classname, so they can be selected
+  // from CSS using that exact value.
+  const root = html`
+    <div class="fill">
+      <div class="fill">
+        <video
+          ref="video"
+          playsinline
+          controlslist="nodownload nofullscreen noremoteplayback"
+        >
+          <!-- Video children are later propagated from the iframe's name. -->
+        </video>
+      </div>
+
+      <div ref="adContainer" class="fill" hidden>
+        <!-- This subtree may be modified by the IMA SDK. -->
+      </div>
+
+      <div ref="controls" hidden>
+        <button ref="playButton"></button>
+
+        <div class="countdownWrapper">
+          <div ref="countdown">
+            <!-- Text content updated using data-ad-label onAdProgress(). -->
+          </div>
+        </div>
+
+        <div ref="time">
+          <!-- Text content must match format in updateTime(). -->
+          -:- / 0:00
+        </div>
+
+        <div ref="progress">
```

Comment:
The less important structural change is that we remove this element's last child (`progress-total-time-line` or sth like that). This element was for presentation only, so we replace its visual effect with the pseudo-child `.progress::after`.